samples/gen-go/client/client.go

```diff
@@ -248,9 +257,12 @@ func (c *WagClient) CreateBook(ctx context.Context, i *models.Book) (*models.Boo
 
 		var err error
 		body, err = json.Marshal(i)
-
 		if err != nil {
-			return nil, err
+
+			if err != nil {
+				return nil, err
+			}
+
```

Comment:
We seem to be doing the same check twice here, which is logically sound but a little weird.